lib/idx/remediate.ts

```diff
@@ -36,26 +35,26 @@ export function getRemediator(
       
     const T = flow[remediation.name];
     remediator = new T(remediation, values);
-    if (remediator.canRemediate()) {
-      // found the remediator
-      return remediator;
-    } else if (flowMonitor.isRemediatorCandidate(remediator, idxRemediations)) {
+    if (flowMonitor.isRemediatorCandidate(remediator, idxRemediations, values)) {
+      if (remediator.canRemediate()) {
+        // found the remediator
+        return remediator;
+      }
       // remediator cannot handle the current values
       // maybe return for next step
-      remediatorCandidates.push(remediator);
+      remediatorCandidates.push(remediator);  
     }
   }
   
-  // TODO: why is it a problem to have multiple remediations?
-  // if (remediatorCandidates.length > 1) {
-  //   const remediationNames = remediatorCandidates.reduce((acc, curr) => {
-  //     const name = curr.getName();
-  //     return acc ? `${acc}, ${name}` : name;
-  //   }, '');
-  //   throw new AuthSdkError(`
-  //     More than one remediation can match the current input, remediations: ${remediationNames}
-  //   `);
-  // }
+  if (remediatorCandidates.length > 1) {
```

Comment:
basically it does not make sense to let the user select another authenticator while doing authenticator verification, here is the logic to tailor the remediations: https://github.com/okta/okta-auth-js/pull/781/files#diff-cc9341a3dda5c9548a9d6f16bbb849abf1c4bf1e5be633c47c8d846db1bec9abR3

Reply:
The user should be able to change their mind and chose a different authenticator.  Maybe I realize I don't have my phone with me, so I want to select email instead.

Reply:
Checked in SIW, looks like there is a `back to authenticators list` for re-select purpose. Then we probably want to return `nextSteps (Array)` instead of `nextStep` in the response.